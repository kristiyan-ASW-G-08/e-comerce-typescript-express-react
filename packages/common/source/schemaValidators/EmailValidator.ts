import * as yup from 'yup';

interface EmailValidatorType {
  email: string;
}
const EmailValidator = yup.object<EmailValidatorType>().shape({
  email: yup
    .string()
    .trim()
    .email()
    .required(),
});
export default EmailValidator;
