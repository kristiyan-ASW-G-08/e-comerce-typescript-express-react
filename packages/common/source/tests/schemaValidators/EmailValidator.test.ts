import EmailValidator from '@schemaValidators/EmailValidator';

describe('EmailValidator', () => {
  const validEmails = [
    'testmail@mail.com',
    'someMail@gmail.com',
    'jojo@test.com',
  ];
  const invalidEmails = ['testmail@mail', 'someMailgmail.com', 'test.com'];
  it.each(validEmails)('should validate successfully', async email => {
    await expect(
      EmailValidator.validate({ email }, { abortEarly: false }),
    ).resolves.toStrictEqual({ email });
  });
  it.each(invalidEmails)('should validate unsuccessfully', async email => {
    await expect(
      EmailValidator.validate({ email }, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
