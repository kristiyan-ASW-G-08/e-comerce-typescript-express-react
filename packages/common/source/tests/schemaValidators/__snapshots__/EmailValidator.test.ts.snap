// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailValidator should validate unsuccessfully 1`] = `[ValidationError: email must be a valid email]`;

exports[`EmailValidator should validate unsuccessfully 2`] = `[ValidationError: email must be a valid email]`;

exports[`EmailValidator should validate unsuccessfully 3`] = `[ValidationError: email must be a valid email]`;
